src/main/kotlin/no/nav/hjelpemidler/delbestilling/plugins/AuthorizationPlugin.kt

```diff
@@ -0,0 +1,35 @@
+package no.nav.hjelpemidler.delbestilling.plugins
+
+import io.ktor.http.HttpStatusCode
+import io.ktor.server.application.createRouteScopedPlugin
+import io.ktor.server.auth.AuthenticationChecked
+import io.ktor.server.response.respond
+import mu.KotlinLogging
+import no.nav.hjelpemidler.delbestilling.AppContext
+import no.nav.hjelpemidler.delbestilling.tokenXUser
+
+private val log = KotlinLogging.logger {}
+
+val DelbestillerRollePlugin = createRouteScopedPlugin(
+    name = "DelbestillerRollePlugin",
+) {
+    pluginConfig.apply {
+        on(AuthenticationChecked) { call ->
+            try {
+                val ctx = AppContext()
+                val rolleService = ctx.rolleService
```

Comment:
Likar idéen! 
Her initialiserar vi AppContext for kvart kall, som ikkje er heilt heldig. Kan vi ta inn AppContext/RolleService via dependency injection? 
Evt. kan vi endre AppContext frå `class AppContext` til `object AppContext`, og skrive `AppContext.rolleService` her (og tilsvarande i Application.kt).